browser/search_engines/android/BUILD.gn

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2025 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+# Import required GN build configuration files
+import("//build/config/android/rules.gni")  # Android build rules
+import("//build/config/locales.gni")
+import("//chrome/common/features.gni")
+import("//tools/grit/grit_rule.gni")
+
+# Define an Android library target for the Quick search engines feature
+android_library("java") {
+  # Java source files to compile
+  sources = [
+    "java/src/org/chromium/brave/browser/SharedPreferencesHelper.java",
```

Comment:
this does not belong here, it's not part of quick search

Reply:
checking with @bridiver in DM.

Reply:
we agreed on having a separate package for `quick_search_engines` so i will be updating the changes accordingly.

Reply:
i have addressed the change here : https://github.com/brave/brave-core/pull/26478/commits/fe9415cbb7ece49697dce100175e6df5c0d95e8a